examples/C++/DDS/CustomListenerExample/CustomListenerPublisher.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CustomListenerPublisher.cpp
+ *
+ */
+
+#include "CustomListenerPublisher.h"
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/PublisherAttributes.h>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+
+#include <thread>
+
+using namespace eprosima::fastdds::dds;
+
+CustomListenerPublisher::CustomListenerPublisher()
+    : participant_(nullptr)
+    , publisher_(nullptr)
+    , topic_(nullptr)
+    , writer_(nullptr)
+    , type_(new HelloWorldPubSubType())
+{
+}
+
+bool CustomListenerPublisher::init(bool use_dw)
+{
+    hello_.index(0);
+    hello_.message("HelloWorld");
+    DomainParticipantQos pqos;
+    pqos.name("Participant_pub");
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(0, pqos, &p_listener_);
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    //REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    //CREATE THE PUBLISHER
+    publisher_ = participant_->create_publisher(PUBLISHER_QOS_DEFAULT, nullptr);
+
+    if (publisher_ == nullptr)
+    {
+        return false;
+    }
+
+    topic_ = participant_->create_topic("HelloWorldTopic", "HelloWorld", TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE WRITER
+    StatusMask none_mask = StatusMask::none();
+    StatusMask all_mask = StatusMask::all();
+    dw_mask_ = use_dw ? all_mask : none_mask;
+    
+    writer_ = publisher_->create_datawriter(topic_, DATAWRITER_QOS_DEFAULT, &dw_listener_, dw_mask_);
```

Comment:
We can avoid the local variables and the member `dw_mask_` altogether.
```suggestion
    writer_ = publisher_->create_datawriter(topic_, DATAWRITER_QOS_DEFAULT, &dw_listener_,
            use_dw ? StatusMask::all() : StatusMask::none());
```